internal/authorization/openfga_authorizer.go

```diff
@@ -0,0 +1,140 @@
+package authorization
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+	openfga "github.com/openfga/go-sdk"
+	"github.com/openfga/go-sdk/client"
+	"github.com/openfga/go-sdk/credentials"
+)
+
+type OpenFGAAuthorizer struct {
+	c *client.OpenFgaClient
+}
+
+func (c *OpenFGAAuthorizer) APIClient() openfga.APIClient {
+	return c.c.APIClient
+}
+
+func WriteOpenFGAAuthorizationModel(ctx context.Context, apiScheme, apiHost, storeId string) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	conf := client.ClientConfiguration{
+		ApiScheme: apiScheme,
+		ApiHost:   apiHost,
+		StoreId:   storeId,
+	}
+
+	fgaClient, err := client.NewSdkClient(&conf)
+	if err != nil {
+		return "", err
+	}
+
+	var builtinAuthorizationModel client.ClientWriteAuthorizationModelRequest
+	err = json.Unmarshal([]byte(authModel), &builtinAuthorizationModel)
+	if err != nil {
+		return "", err
+	}
+
+	data, err := fgaClient.WriteAuthorizationModel(ctx).Body(builtinAuthorizationModel).Execute()
+	if err != nil {
+		return "", err
+	}
+
+	return data.GetAuthorizationModelId(), nil
+}
```

Comment:
i'd chuck this on another object as it wont be really used by this codebase until we have a separate `app migrate` command but at that point we might want to separate the logic too